aloha-core/src/main/scala/com/eharmony/aloha/semantics/compiled/plugin/schemabased/SchemaBasedSemanticsPlugin.scala

```diff
@@ -0,0 +1,228 @@
+package com.eharmony.aloha.semantics.compiled.plugin.schemabased
+
+import com.eharmony.aloha.semantics.compiled.plugin.schemabased.accessor._
+import com.eharmony.aloha.semantics.compiled.plugin.schemabased.codegen.CodeGenerators
+import com.eharmony.aloha.semantics.compiled.plugin.schemabased.codegen.MapType.{FLAT_MAP, MAP, NONE}
+import com.eharmony.aloha.semantics.compiled.plugin.schemabased.schema.Schema.FieldRetrievalError
+import com.eharmony.aloha.semantics.compiled.plugin.schemabased.schema.{FieldDesc, ListField, RecordField, Schema}
+import com.eharmony.aloha.semantics.compiled.plugin.schemabased.tokenization.{Field, Index, SchemaBasedTokenizer, Token}
+import com.eharmony.aloha.semantics.compiled.{CompiledSemanticsPlugin, OptionalAccessorCode, RequiredAccessorCode, VariableAccessorCode}
+
+import scala.annotation.tailrec
+import scalaz.ValidationNel
+import scalaz.syntax.validation.ToValidationV
+
+/**
+  * Created by ryan.deak on 2/21/17.
+  */
+trait SchemaBasedSemanticsPlugin[A] { self: CompiledSemanticsPlugin[A] =>
+
+  private[this]type FieldAccessorPartition = (List[FieldAccessor], Option[Repeated], List[FieldAccessor])
+  private[this]type MappingPartition = (List[(Seq[Req], Opt)], Option[(Seq[Req], Opt)], List[Req])
+
+  private[schemabased] val functionParamList = "(_0: " + inputTypeString + ") => "
+
+  def dereferenceAsOptional: Boolean
+  protected[this] def schema: Schema
+  protected[this] def codeGenerators: CodeGenerators
+
+  def accessorFunctionCode(spec: String): Either[Seq[String], VariableAccessorCode] = {
+    val function = for {
+      tokens <- SchemaBasedTokenizer.getTokens(spec)
+      subfields <- fieldChain(fieldNames(tokens), schema)
+      fap <- getFieldAccessorPartition(subfields, tokens)
```

Comment:
I had to look up the definition but I don't want to change based on some other connotation.